virtualizarr/tests/test_integration.py

```diff
@@ -302,6 +326,93 @@ def test_datetime64_dtype_fill_value(
+        time_vars: list[str],
+    ):
+        if roundtrip_func is not roundtrip_as_in_memory_icechunk:
+            pytest.xfail("not implemented")
+
+        # set up example xarray dataset
+        with xr.tutorial.open_dataset(
+            "air_temperature", decode_times=decode_times
+        ) as ds:
+            # split into two datasets
+            ds1 = ds.isel(time=slice(None, 1460))
+            ds2 = ds.isel(time=slice(1460, None))
+
+            # save it to disk as netCDF (in temporary directory)
+            air1_nc_path = tmp_path / "air1.nc"
+            air2_nc_path = tmp_path / "air2.nc"
+            ds1.to_netcdf(air1_nc_path)
+            ds2.to_netcdf(air2_nc_path)
+
+            # use open_dataset_via_kerchunk to read it as references
+            with (
+                open_virtual_dataset(
+                    str(air1_nc_path),
+                    loadable_variables=time_vars,
+                    decode_times=decode_times,
+                    backend=hdf_backend,
+                ) as vds1,
+                open_virtual_dataset(
+                    str(air2_nc_path),
+                    loadable_variables=time_vars,
+                    decode_times=decode_times,
+                    backend=hdf_backend,
+                ) as vds2,
+            ):
+                if not decode_times:
+                    assert vds1.time.dtype == np.dtype("float32")
+                else:
+                    assert vds1.time.dtype == np.dtype("<M8[ns]")
+                    assert "units" in vds1.time.encoding
+                    assert "calendar" in vds1.time.encoding
+
+                vdt = xr.DataTree.from_dict({"/vds1": vds1, "/vds2": vds2})
+
+                with roundtrip_func(
+                    vdt, tmp_path, decode_times=decode_times
+                ) as roundtrip:
+                    assert isinstance(roundtrip, xr.DataTree)
+
+                    if not decode_times:
+                        # assert all_close to original dataset
+                        roundtrip_vds1 = roundtrip["vds1"].to_dataset()
+                        roundtrip_vds2 = roundtrip["vds2"].to_dataset()
+                        xrt.assert_allclose(roundtrip_vds1, ds1)
+                        xrt.assert_allclose(roundtrip_vds2, ds2)
+
+                        # assert coordinate attributes are maintained
+                        for coord in ds1.coords:
+                            assert (
+                                ds1.coords[coord].attrs
+                                == roundtrip_vds1.coords[coord].attrs
+                            )
+                        for coord in ds2.coords:
+                            assert (
+                                ds2.coords[coord].attrs
+                                == roundtrip_vds2.coords[coord].attrs
+                            )
+                    else:
+                        # they are very very close! But assert_allclose doesn't seem to work on datetimes
```

Comment:
Is this comment copied from elsewhere or did you discover this?

Reply:
I copied this test function from `test_kerchunk_roundtrip_concat` in the same file (which includes the very same comment), and adjusted for testing DataTree roundtrip, but with only the "in memory" roundtrip function.

However, since reading your comment here, I see that the comment is no longer accurate in either test function. That is, `assert_allclose` does work on datetimes, so I have adjusted both tests accordingly.